external/types/web.d.ts

```diff
@@ -0,0 +1,924 @@
+    update(scale?: number, rotation?: number): void;
+    cancelRendering(keepAnnotations?: boolean): void;
+    cssTransform(
+      target: HTMLCanvasElement | SVGElement,
+      redrawAnnotations?: boolean
+    ): void;
+    width: number;
+    height: number;
+    getPagePoint(x: number, y: number): Point;
+    draw(): Promise<void>;
+    paintOnCanvas(canvasWrapper: HTMLDivElement): RenderTask;
+    paintOnSvg(wrapper: HTMLDivElement): RenderTask;
+    /**
+     * @param {string|null} label
+     */
+    setPageLabel(label: string | null): void;
+  }
+
+  //
+  // web/pdf_viewer.js
+  //
+  class PDFViewer extends BaseViewer {
+    update(): void;
+  }
+
+  //
+  // web/pdf_thumbnail_viewer.js
+  //
+
+  /**
+   * @typedef {Object} PDFThumbnailViewerOptions
+   * @property {HTMLDivElement} container - The container for the thumbnail
+   *   elements.
+   * @property {IPDFLinkService} linkService - The navigation/linking service.
+   * @property {PDFRenderingQueue} renderingQueue - The rendering queue object.
+   * @property {IL10n} l10n - Localization service.
+   */
+  interface PDFThumbnailViewerOptions {
+    container: HTMLDivElement;
+    linkService: IPDFLinkService;
+    renderingQueue: PDFRenderingQueue;
+    l10n: IL10n;
+  }
+
+  class PDFThumbnailViewer {
+    container: HTMLDivElement;
+    linkService: IPDFLinkService;
+    renderingQueue: PDFRenderingQueue;
+    l10n: IL10n;
+    scroll: {
+      down: boolean;
+      lastY: number;
+      _eventHandler: (evt: any) => void;
+    };
+
+    constructor(options: PDFThumbnailViewerOptions);
+    getThumbnail(index: PageIndex): PDFThumbnailView;
+    scrollThumbnailIntoView(page: PageIndex): void;
+    pagesRotation: number;
+    cleanup(): void;
+    setDocument(pdfDocument: PDFDocumentProxy): void;
+    setPageLabels(labels: string[]): void;
+    forceRendering(): boolean;
+  }
+
+  /**
+   * @typedef {Object} PDFThumbnailViewOptions
+   * @property {HTMLDivElement} container - The viewer element.
+   * @property {number} id - The thumbnail's unique ID (normally its number).
+   * @property {PageViewport} defaultViewport - The page viewport.
+   * @property {IPDFLinkService} linkService - The navigation/linking service.
+   * @property {PDFRenderingQueue} renderingQueue - The rendering queue object.
+   * @property {boolean} disableCanvasToImageConversion - (optional) Don't convert
+   *   the canvas thumbnails to images. This prevents `toDataURL` calls,
+   *   but increases the overall memory usage. The default value is `false`.
+   * @property {IL10n} l10n - Localization service.
+   */
+  interface PDFThumbnailViewOptions {
+    container: HTMLDivElement;
+    id: PageIndex;
+    defaultViewport: PageViewport;
+    linkService: IPDFLinkService;
+    renderingQueue: PDFRenderingQueue;
+    disableCanvasToImageConversion?: boolean;
+    l10n: IL10n;
+  }
+
+  //
+  // web/pdf_thumbnail_view.js
+  //
+  class PDFThumbnailView {
```

Comment:
This isn't exposed, please see https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.component.js for a list of the viewer components that are.